File: openapi-test-java-server-jaxrs/testapi/model/Owner.java
package testapi.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotNull;
import java.util.Optional;

public record Owner(
  @JsonProperty("address") Optional<Address> address,
  @JsonProperty("email") @jakarta.validation.constraints.Email Optional<String> email,
  @JsonProperty("id") @NotNull String id,
  @JsonProperty("name") @NotNull String name
) {
  public Owner withAddress(Optional<Address> address) {
    return new Owner(address, email, id, name);
  };

  public Owner withEmail(Optional<String> email) {
    return new Owner(address, email, id, name);
  };

  public Owner withId(String id) {
    return new Owner(address, email, id, name);
  };

  public Owner withName(String name) {
    return new Owner(address, email, id, name);
  };
}

File: openapi-test-kotlin-server-jaxrs-client-mp/testapi/model/Owner.kt
package testapi.model

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotNull
import java.util.Optional

data class Owner(
  @field:JsonProperty("address") val address: Optional<Address>,
  @field:JsonProperty("email") @jakarta.validation.constraints.Email val email: Optional<String>,
  @field:JsonProperty("id") @NotNull val id: String,
  @field:JsonProperty("name") @NotNull val name: String
)